zallet/src/components/json_rpc/methods/view_transaction.rs

```diff
@@ -302,9 +323,10 @@ pub(crate) async fn call(
             .with_raw(|conn| {
                 conn.query_row(
                     &format!(
-                        "SELECT txid, {output_prefix}_index, address, value
-                        FROM {pool_prefix}_received_notes
+                        "SELECT txid, {output_prefix}_index, accounts.uuid, address, value
+                        FROM {pool_prefix}_received_notes rn
                         JOIN transactions ON tx = id_tx
+                        JOIN accounts ON accounts.id = rn.account_id
```

Comment:
This comment isn't really specific to this PR, but I think that taking the approach of looking up the spend by the nullifier runs a small risk of producing inconsistencies between different views in the wallet, because that's not how we look up spends anywhere else. The way I would have approached that particular issue (in the earlier PR, where I thought about this but didn't want to take time asking for changes on) would be to query the `v_received_note_spends` view to find all the outputs known by the wallet to be spent in the transaction in question.